pkg/clusterversion/clusterversion.go

```diff
@@ -90,3 +98,248 @@ func (builder *Builder) validate() (bool, error) {
 
 	return true, nil
 }
+
+// WithDesiredUpdateImage adds the desired image to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateImage(desiredUpdateImage string, force bool) (*Builder, error) {
+	var err error
+
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if desiredUpdateImage == "" {
+		glog.V(100).Infof("The desiredUpdateImage is empty")
+
+		return builder, fmt.Errorf("'desiredUpdateImage' cannot be empty")
+	}
+
+	glog.V(100).Info("Updating clusterversion %s to the desired image %s",
+		builder.Definition.Name, desiredUpdateImage)
+
+	builder.Definition.Spec.DesiredUpdate = &v1.Update{Image: desiredUpdateImage, Force: force}
+
+	return builder, err
+}
+
+// WithDesiredUpdateChannel adds the desired channel to the clusterversion struct.
+func (builder *Builder) WithDesiredUpdateChannel(updateChannel string) (*Builder, error) {
+	var err error
+
+	if valid, err := builder.validate(); !valid {
+		return builder, err
+	}
+
+	if updateChannel == "" {
+		glog.V(100).Infof("The updateChannel is empty")
+
+		return builder, fmt.Errorf("'updateChannel' cannot be empty")
+	}
+
+	glog.V(100).Info("Updating clusterversion %s to the desired updateChannel %s",
```

Comment:
Please move this log after validate. It'll add more viability in logs. Change Updating to Adding

Reply:
Moved, thanks